### PR TITLE
docs: create `commonTransactionParameters` to hold parameters that are common between all transaction types

### LINKED ISSUE #208: Add documentation for a common JSON transaction parameters object
An issue was noticed while working on https://github.com/hashgraph/hedera-sdk-cpp/issues/695 (with #205) that it isn't exactly clear when an additional signature should be applied to a transaction for some tests. The use of the `signerKey` field should be used to explicitly provide the keys that should sign a transaction.

### PR DESCRIPTION
**Pull request by rwalworth**
~~- Update `signerKey` to a list that will allow for multiple additional signatures of a transaction~~
~~- Explicitly use `signerKeys` to provide any additional signatures to a transaction. The transaction will not be signed (except by the fee payer) if `signerKeys` is empty.~~

While working on #216, I noticed an issue that would need to be addressed at some point: common parameters for all transactions. I decided to hijack this PR with a proposed solution since it was already addressing an issue that was tangent to it. Basically in order to allow ourselves to test all fields for a transaction (most notably, the parameters that are common to all transactions), we would need to add all of these parameters 

This PR now adds the specification for an additional JSON parameter object that can be shared between transactions, `commonTransactionParams`. This will allow every transaction to hold common parameters in one defined object, as well as hold signatures (which was this PR's original intent). `signerKey` is now `signers`, and is a list contained in `commonTransactionParams`. The transaction will now not be signed (except by the fee payer established with the `setOperator` method) if `commonTransactionParams.signers` is empty.

**Related issue(s)**:

Fixes #208

### CODE REVIEW

File: test-specifications/crypto-service/accountCreateTransaction.md [hunk 1]
```diff
@@ -136,11 +136,11 @@ https://docs.hedera.com/hedera/sdks-and-apis/rest-api
...
+|---------|--------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------|-------------------|
+| 1       | Creates an account that requires a receiving signature                                     | key=<VALID_PUBLIC_KEY>, receiverSignatureRequired=true, signerKeys=[<ASSOCIATED_PRIVATE_KEY>] | The account creation succeeds and the account requires a receiving signature.        | N                 |
+| 2       | Creates an account that doesn't require a receiving signature                              | key=<VALID_PRIVATE_KEY>, receiverSignatureRequired=false                                      | The account creation succeeds and the account doesn't require a receiving signature. | N                 |
```
[the remark this answers is not in the document]
Comment (rwalworth) [reply test-specifications/crypto-service/accountCreateTransaction.md]: Nope good catch, any `VALID_KEY` should be good. Updated

**Review by thenswan — APPROVED** [review]
gg @rwalworth

**Review by thenswan — APPROVED** [review]
I guess it makes sense, although would like to have a review here from other team members as well.

**Comment by thenswan** [comment]
@rwalworth will `AccountCreateTransaction` be updated with `commonTransactionParams`?

**Comment by rwalworth** [comment]
> @rwalworth will `AccountCreateTransaction` be updated with `commonTransactionParams`?

Yes, the `createAccount` JSON-RPC method will be updated to contain this (#205 and #206)

**Pull request merged**